react/features/visitors/middleware.ts

```diff
@@ -46,28 +52,62 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
         const { conference } = action;
 
         if (getState()['features/visitors'].iAmVisitor) {
-            dispatch(showNotification({
+            const { demoteActorDisplayName } = getState()['features/visitors'];
```

Comment:
I am wondering isn't it better to show the notification as soon as we start the demote process? This way even if we fail to join for some reason we would have already displayed the notification and the user will know what is happening. You will also not need to store the demote actor display name to redux and you will be able to use it right away. WDYT?